### ISSUE TITLE
Linux: Wire Desktop steals the focus from other windows when receiving a chat message

### ISSUE THREAD

**Issue by pkolmann** [wireapp/wire-desktop#700]
Wire Version: 2.13.2741
Wire for Web Version: 2017.06.07.1805
OS: Debian SID
Do you have an antivirus software installed: No

What steps will reproduce the problem?
1. open wire-desktop
2. go to a different window and let someone send you a message
3. wire-desktop app gets the focus and whatever you type, you continue to type in the wire chat window. That extremely bad if you are currently typing a password....

What is the expected result?
leave the focus alone.

What happens instead?
wire grabs the focus

**Comment by ConorIA** [comment]
I have not experienced this before. On a new message you should just see a notification. Unless I'm mistaken the window only comes into focus if you click that notification. In fact other users asked for Wire to appear automatically on a new message,  but the devs declined the request because it would be too intrusive.

**Comment by ffflorian** [comment]
Same here, cannot reproduce on Debian. @pkolmann which window manager do you use?

**Comment by pkolmann** [comment]
I have fvwm running as window manager. It seems to be an issue with the node.js server. I have the same issue with the new skype4linx.

Package: fvwm
Version: 1:2.6.7-3

**Comment by ffflorian** [comment]
@pkolmann 
Thanks for the answer. Since this is not a Wire specific issue, I'll close this one.

**Comment by pkolmann** [comment]
I have checked with FVWM2 docu and found the following which solved the issue for me. (Adding here, if anyone else comes across this issue):

`# prevent focus stealing
# see http://fvwmforums.org/wiki/Tips/FocusStealing/

DestroyFunc EWMHActivateWindowFunc
DestroyFunc UrgencyFunc
`

**Comment by jaywink** [comment]
I have this exact issue. Debian package install (from releases).

$ dpkg -l wire-desktop
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version        Architecture Description
+++-==============-==============-============-=================================
ii  wire-desktop   3.16.2923-2923 amd64

$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.1 LTS"

My window manager is the Ubuntu default GNOME shell.

I don't think this was always the case (the focus stealing), but can't be sure. It's definitely been happening for some months at least.

Anything I can do to produce some more information?

**Comment by ffflorian** [comment]
@jaywink please install the latest version (3.19) and see if the problem persists.

**Comment by jaywink** [comment]
@ffflorian I just realized that my version is not the latest. Which is odd since Wire has told me it has an update several times in the app itself and I assumed it updates itself?

I just installed the latest version and the problem persists.

wire-desktop   3.19.2928-2928 amd64

**Comment by ffflorian** [comment]
> Which is odd since Wire has told me it has an update several times in the app itself and I assumed it updates itself?

Yeah, the update message is a bit misleading, sorry about that. We have an [open issue](https://github.com/wireapp/wire-desktop/issues/3802) about that.

Please explain exactly how the Wire window behaves and how it should behave in your opinion, since I can't reproduce it. Thanks!

**Comment by jaywink** [comment]
OK a bit more info now that I started playing around different modes. If the window is closed, there is only a notification. If the window is open but behind another window, it gets focus on receiving a message,
